src/js/apps/patients/worklist/filters_app.js

```diff
@@ -64,41 +58,26 @@ export default App.extend({
 
     this.showChildView('toggle', typeToggleView);
   },
-  showCliniciansFilterView() {
-    const selected = Radio.request('entities', 'clinicians:model', this.getState('clinicianId'));
-
-    const clinicianFilter = new ClinicianDropList({
-      groups: this.groups,
-      state: { selected },
-    });
-
-    this.listenTo(clinicianFilter, 'change:selected', ({ id }) => {
-      this.setState('clinicianId', id);
-    });
-
-    this.showChildView('clinician', clinicianFilter);
-  },
-  showRolesFilterView() {
-    const role = Radio.request('entities', 'roles:model', this.getState('roleId'));
-
-    const roleFilter = new RoleComponent({ role });
-
-    this.listenTo(roleFilter, 'change:selected', ({ id }) => {
-      this.setState('roleId', id);
-    });
-
-    this.showChildView('role', roleFilter);
-  },
-  showOwnerFilterView() {
-    const owner = this.getState('clinicianId') ?
-      Radio.request('entities', 'clinicians:model', this.getState('clinicianId')) :
-      Radio.request('entities', 'roles:model', this.getState('roleId'));
+  showOwnerFilterView(hasRoles = true, hasClinicians = true) {
+    // first render: has both role id and clinician id
+    // is role if both hasRoles and hasClinicians is true
+    let owner;
+
+    if (this.getState('clinicianId') && this.getState('roleId') && hasRoles && !hasClinicians) {
```

Comment:
Would this not work? `const owner = (!hasClinicians || !this.getState('clinicianId'))? <request role> : <request clinician>;`